tests/manage/mcg/test_non_ocs_taint_and_toleration.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import pytest
+
+from ocs_ci.ocs import ocp, constants, defaults
+
+from ocs_ci.framework.testlib import (
+    E2ETest,
+    ignore_leftovers,
+    skipif_tainted_nodes,
+    skipif_managed_service,
+)
+from ocs_ci.ocs.resources.pod import (
+    get_all_pods,
+    wait_for_pods_to_be_running,
+    check_toleration_on_pods,
+)
+from ocs_ci.ocs.node import (
+    taint_nodes,
+    untaint_nodes,
+    get_worker_nodes,
+)
+from ocs_ci.framework.pytest_customization.marks import bugzilla
+from ocs_ci.helpers.sanity_helpers import Sanity
+
+logger = logging.getLogger(__name__)
+
+
+@ignore_leftovers
+@skipif_tainted_nodes
+@skipif_managed_service
+@bugzilla("2115613")
+class TestNonOCSTaintAndTolerations(E2ETest):
+    """
+    Test to test non ocs taints on ocs nodes
+    and toleration
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+        """
+        Make sure all nodes are untainted
+
+        """
+
+        def finalizer():
+            assert untaint_nodes(
+                taint_label="xyz=true:NoSchedule",
+            ), "Failed to untaint"
+
```

Comment:
To fully revert to the previous state we should also remove the tolerations from the storagecluster and the subscription. Afterward, we should also respin the pods so that the new configuration will propagate to the pods.